pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -804,6 +804,7 @@ func (c *Controller) updateEDS(ep *v1.Endpoints, event model.Event) {
 						UID:             uid,
 						ServiceAccount:  secureNamingSAN(pod),
 						Network:         c.endpointNetwork(ea.IP),
+						Locality:        c.GetPodLocality(pod),
```

Comment:
As i have commented, this function is not efficient.

Reply:
@hzxuzhonghu Would you mind being more explicit or at least giving a link to your other comment, please?

Reply:
https://github.com/istio/istio/pull/14464#discussion_r289672660

The problem is in get pod locality we first determine its real locality, then if there is an override we throw away the real locality. Slightly more efficient would be to only compute the locality if the override is not present.

But getting the real locality is not expensive, so I think it is more of a micro-optimization.